lib/mayaUsd/resources/scripts/mayaUsdAddMayaReference.mel

```diff
@@ -455,9 +753,48 @@ global proc addMayaReferenceToUsdUiCb(string $parent, string $selectedFile)
     global int $gLastOptionSelectionIndex;
     global string $gLastOptionSelection;
     global string $gCurrentOptionSelection;
+
     string $fileType;
     string $optionsScript;
 
+    // Save the various Usd Maya Reference Options.
+    int $defineInVar = `checkBoxGrp -q -v1 usdMayaReferenceDefineInVariantGrp`;
+    optionVar -iv usdMayaReferenceDefineInVariant $defineInVar;
+
+    if ($defineInVar)
+    {
+        string $createNew = getMayaUsdLibString("kMayaRefCreateNew");
+        string $varSetNameText = `textField -q -text usdMayaReferenceVariantSetNameTextField`;
+        if (`optionMenu -q -ni usdMayaReferenceVariantSetNameOptionMenu` > 0) {
+            // If "Create New" is used, then set the optionVar for the optionMenu to the value
+            // from the textField (the new Variant Set name that will be created).
+            string $varSetName = `optionMenu -q -value usdMayaReferenceVariantSetNameOptionMenu`;
+            if ($varSetName == $createNew) {
+                optionVar -sv usdMayaReferenceVariantSetNameOption $varSetNameText;
+            } else {
+                optionVar -sv usdMayaReferenceVariantSetNameOption $varSetName;
+            }
+        } else {
+            optionVar -sv usdMayaReferenceVariantSetNameOption $varSetNameText;
+        }
+        optionVar -sv usdMayaReferenceVariantSetNameText $varSetNameText;
```

Comment:
Sorry another change here regarding the optionVar values after a create new (spec update from Natalia).